src/main/java/com/pinterest/secor/common/OffsetTracker.java

```diff
@@ -52,15 +52,15 @@ public long getLastSeenOffset(TopicPartition topicPartition) {
     public long setLastSeenOffset(TopicPartition topicPartition, long offset) {
         long lastSeenOffset = getLastSeenOffset(topicPartition);
         mLastSeenOffset.put(topicPartition, offset);
-        if (lastSeenOffset + 1 != offset) {
-            if (lastSeenOffset >= 0) {
-                LOG.warn("offset for topic {} partition {} changed from {} to {}",
-                        topicPartition.getTopic(),topicPartition.getPartition(),lastSeenOffset, offset);
-            } else {
+        if (offset < lastSeenOffset + 1) {
+            LOG.warn("offset for topic {} partition {} goes back from {} to {}",
```

Comment:
How about we throw an exception here?  I don't think we should see this happen with the code change in Rebalance listener?

Reply:
For compatibility with LegacyKafkaMessageIterator, no such listener exists, so we have to keep this for now.

Reply:
Would prefer to use a separator class for RebalanceListener since rebalance logic is separator concern from message iterator including upload and cleanup.

Reply:
Let me know if the above is ok, will push shortly.